src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -246,7 +246,9 @@ trait LowPriorityConfigConvertImplicits {
 
   implicit val readString = fromString[String](identity)
   implicit val readBoolean = fromString[Boolean](_.toBoolean)
-  implicit val readDouble = fromString[Double](_.toDouble)
+  implicit val readDouble = fromString[Double]({
+    v => if (v.last == '%') v.dropRight(1).toDouble / 100.0 else v.toDouble
```

Comment:
I added the `fromNonEmptyString` method in eb3ffb1b2487c8dbd233e2221f6bfcde0cacdd86. However, instead of requiring an explicit typename, I've added a context bound for `ClassTag`, in order to extract the typename.